src/ECL/ECL.hs

```diff
@@ -31,6 +35,15 @@ data Binder lit
   | Record [(Name, Binder lit)]
   deriving (Show, Eq)
 
+-- | Represents constructors' arities.
+type Arity = Int
+
+-- | Environment
+--
+-- The language implementor should provide an environment to let
+-- the checker lookup for constructors' names (just for one type for now).
+type Environment = [(Name, Arity)]
```

Comment:
I would make this into a `newtype`, with a record accessor:

```haskell
newtype Environment = Environment { envArities :: [(Name, Arity)] }
```

and provide a smart constructor

```haskell
env :: [(Name, Arity)] -> Environment
```

or a default value:

```haskell
defaultEnv :: Environment
```

so that we can add things we need here later, without breaking the API.